pkg/ottl/ottlfuncs/func_replace_all_matches.go

```diff
@@ -43,9 +45,22 @@ func replaceAllMatches[K any](target ottl.PMapGetter[K], pattern string, replace
 		if err != nil {
 			return nil, err
 		}
-		replacementVal, err := replacement.Get(ctx, tCtx)
-		if err != nil {
-			return nil, err
+		if fn.IsEmpty() {
+			replacementVal, err = replacement.Get(ctx, tCtx)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			fnVal := fn.Get()
+			replacementExpr, errNew := fnVal.Get(&FuncArgs[K]{Input: replacement})
```

Comment:
```suggestion
			replacementExpr, err := fnVal.Get(&FuncArgs[K]{Input: replacement})
```
Is there a reason we can't just use `err` here? It looks like we immediately return on seeing a non-nil error in this function.

Reply:
I run into this issue - `shadow: declaration of "err" shadows declaration in the outer scope`